### PR TITLE
chore: update colors

### PR DESCRIPTION
**Pull request by janndriessen**
* Updates colors in app to use tailwind config (preferred color system)
* Updates tailwind config to be in [typescript](https://tailwindcss.com/docs/configuration#using-esm-or-type-script)
* Updates tailwind config position in project (moved from _root_ to `/lib/styles`)

###### _By submitting this pull request, you are confirming the following to be true:_

- I have reviewed the [Contribution Guidelines](https://github.com/IndexCoop/index-app/blob/master/CONTRIBUTING.md).
- I have performed a self-review of my own code.
- I have updated my repository to match the master at `IndexCoop/index-app`.
- I have included test data or screenshots that prove my fix is effective or that my feature works.

### CODE REVIEW

File: src/components/supply/index.tsx [hunk 1]
```diff
@@ -92,7 +92,7 @@ const TitleAndDescription = () => {
   return (
     <Flex direction={'column'}>
       <Flex align='center'>
-        <InfoIcon color={colors.icBlue4} height='20px' width='20px' />
+        <InfoIcon color={theme.colors.blue[900]} height='20px' width='20px' />
```
Comment (janndriessen) [inline src/components/supply/index.tsx:95]: TODO: can probably use className here

File: src/components/supply/reth-supply-cap.tsx [hunk 2]
```diff
@@ -9,20 +9,21 @@ interface RethSupplyCapProps {
...
+  const { theme } = useTheme()
   if (progress === 100) {
-    return colors.icBlue5
+    return theme.colors.blue[950]
   }
   if (progress > 100) {
     return colors.icRed
   }
-  return colors.icBlue4
+  return theme.colors.blue[900]
```
Comment (janndriessen) [inline src/components/supply/reth-supply-cap.tsx:20]: Needs solution to statically read colors (on the server side)

File: src/app/providers.tsx [hunk 3]
```diff
@@ -23,7 +23,7 @@ export function Providers({ children }: { children: React.ReactNode }) {
     id: process.env.NEXT_PUBLIC_GOOGLE_TAG_MANAGER_CONTAINER_ID ?? '',
   }
   return (
-    <CacheProvider>
+    <CacheProvider prepend={true}>
```
Comment (janndriessen) [inline src/app/providers.tsx:26]: With this there should be less issues between chakra <> tailwind - as chakra should be loaded first and tailwind styles are applied afterwards.

File: src/components/swap/components/select-token-modal.tsx [hunk 4]
```diff
@@ -101,42 +101,49 @@ const TokenItem = ({
...
+          </Text>
         </Flex>
-        <Text fontSize='sm' fontWeight='500' textColor={colors.icBlack}>
-          {item.name}
+      </Flex>
+      <Flex align='flex-end' direction='column' ml='16px'>
+        <Text fontSize='md' fontWeight='700' textColor={colors.ic.black}>
+          {balance}
         </Text>
+        <Text fontSize='sm' fontWeight='500'></Text>
```
Comment (0xonramp) [inline src/components/swap/components/select-token-modal.tsx:145]: Is this used for anything visually, or can remove?

**Review by 0xonramp — APPROVED** [review]
One smol question (element appears to have existed before as well)

File: src/components/swap/components/select-token-modal.tsx (hunk 4 above)
Comment (janndriessen) [reply src/components/swap/components/select-token-modal.tsx]: Removed.

**Pull request merged**